marimo/_server/api/middleware.py

```diff
@@ -408,13 +408,30 @@ async def __call__(
             content=request.stream(),
         )
 
-        rp_resp = await client.send(rp_req, stream=True)
-        response = StreamingResponse(
-            rp_resp.aiter_raw(),
-            status_code=rp_resp.status_code,
-            headers=rp_resp.headers,
-            background=BackgroundTask(rp_resp.aclose),
-        )
+        try:
+            rp_resp = await client.send(rp_req, stream=True)
+            response = StreamingResponse(
+                rp_resp.aiter_raw(),
+                status_code=rp_resp.status_code,
+                headers=rp_resp.headers,
+                background=BackgroundTask(rp_resp.aclose),
+            )
+        except ConnectionRefusedError:
+            # Check if this is a matplotlib server request (contains /mpl/ in path)
+            if "/mpl/" in request.url.path:
```

Comment:
feels weird to have this mpl specific edge case in the proxy middleway.

could we instead either extend `ProxyMiddleware` to `MplProxyMiddleware` or pass down a set of arguments to support this?

Reply:
Generalized to a handler